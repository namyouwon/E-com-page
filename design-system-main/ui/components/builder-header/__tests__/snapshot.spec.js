// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

/* eslint-env jest */

import React from 'react';
import createHelpers from '../../../../jest.helpers';
import BuilderHeader from '../';
const { matchesMarkup } = createHelpers(__dirname);

describe('render builder header', () => {
  it('render base builder header', () => matchesMarkup(<BuilderHeader />));
  it('render base builder header with truncated name', () =>
    matchesMarkup(
      <BuilderHeader docName="Page Type with a very very long name that will truncate when the container is not wide enough to contain this content completely" />
    ));
  it('render base builder header with toolbar', () =>
    matchesMarkup(<BuilderHeader showToolbar />));
  it('render base builder header with toolbar and status with alert', () =>
    matchesMarkup(<BuilderHeader hasStatus hasWarning hasError showToolbar />));
});
