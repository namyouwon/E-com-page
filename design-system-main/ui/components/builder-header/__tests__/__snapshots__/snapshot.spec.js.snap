// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render builder header render base builder header 1`] = `
<div
  className="slds-builder-header_container"
>
  <header
    className="slds-builder-header"
  >
    <div
      className="slds-builder-header__item"
    >
      <a
        className="slds-builder-header__item-action"
        href="#"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        title="Back"
      >
        <span
          className="slds-icon_container slds-icon-utility-back slds-current-color"
          title={null}
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon_x-small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#back"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Back
          </span>
        </span>
      </a>
    </div>
    <div
      className="slds-builder-header__item"
    >
      <div
        className="slds-builder-header__item-label slds-media slds-media_center"
      >
        <div
          className="slds-media__figure"
        >
          <span
            className="slds-icon_container slds-icon-utility-builder slds-current-color"
            title={null}
          >
            <svg
              aria-hidden={true}
              className="slds-icon slds-icon_x-small"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#builder"
              />
            </svg>
          </span>
        </div>
        <div
          className="slds-media__body"
        >
          App Name
        </div>
      </div>
    </div>
    <nav
      className="slds-builder-header__item slds-builder-header__nav"
    >
      <ul
        className="slds-builder-header__nav-list"
      >
        <li
          className="slds-builder-header__nav-item slds-dropdown-trigger slds-dropdown-trigger_click slds-is-open"
        >
          <button
            aria-expanded={false}
            aria-haspopup="true"
            className="slds-button slds-builder-header__item-action slds-media slds-media_center"
            title="Click to open menu"
          >
            <span
              className="slds-media__figure"
            >
              <span
                className="slds-icon_container slds-icon-utility-page slds-current-color"
                title={null}
              >
                <svg
                  aria-hidden={true}
                  className="slds-icon slds-icon_x-small"
                >
                  <use
                    xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#page"
                  />
                </svg>
              </span>
            </span>
            <span
              className="slds-media__body"
            >
              <span
                className="slds-truncate"
                title="Dropdown"
              >
                Dropdown
              </span>
              <span
                className="slds-icon_container slds-icon-utility-chevrondown slds-current-color slds-m-left_small"
                title={null}
              >
                <svg
                  aria-hidden={true}
                  className="slds-icon slds-icon_x-small"
                >
                  <use
                    xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevrondown"
                  />
                </svg>
              </span>
            </span>
          </button>
        </li>
      </ul>
    </nav>
    <div
      className="slds-builder-header__item slds-has-flexi-truncate"
    >
      <h1
        className="slds-builder-header__item-label"
      >
        <span
          className="slds-truncate"
          title="Page Type"
        >
          Page Type
        </span>
      </h1>
    </div>
    <div
      className="slds-builder-header__item slds-builder-header__utilities"
    >
      <div
        className="slds-builder-header__utilities-item"
      >
        <a
          className="slds-builder-header__item-action slds-media slds-media_center"
          href="#"
          onClick={[Function]}
        >
          <span
            className="slds-media__figure"
          >
            <span
              className="slds-icon_container slds-icon-utility-settings slds-current-color"
              title={null}
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon_x-small"
              >
                <use
                  xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
                />
              </svg>
            </span>
          </span>
          <span
            className="slds-media__body"
          >
            <span
              className="slds-truncate"
              title="Link"
            >
              Settings
            </span>
          </span>
        </a>
      </div>
      <div
        className="slds-builder-header__utilities-item"
      >
        <a
          className="slds-builder-header__item-action slds-media slds-media_center"
          href="#"
          onClick={[Function]}
        >
          <div
            className="slds-media__figure"
          >
            <span
              className="slds-icon_container slds-icon-utility-help slds-current-color"
              title={null}
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon_x-small"
              >
                <use
                  xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#help"
                />
              </svg>
            </span>
          </div>
          <div
            className="slds-media__body"
          >
            Help
          </div>
        </a>
      </div>
    </div>
  </header>
</div>
`;

exports[`render builder header render base builder header with toolbar 1`] = `
<div
  className="slds-builder-header_container"
>
  <header
    className="slds-builder-header"
  >
    <div
      className="slds-builder-header__item"
    >
      <a
        className="slds-builder-header__item-action"
        href="#"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        title="Back"
      >
        <span
          className="slds-icon_container slds-icon-utility-back slds-current-color"
          title={null}
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon_x-small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#back"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Back
          </span>
        </span>
      </a>
    </div>
    <div
      className="slds-builder-header__item"
    >
      <div
        className="slds-builder-header__item-label slds-media slds-media_center"
      >
        <div
          className="slds-media__figure"
        >
          <span
            className="slds-icon_container slds-icon-utility-builder slds-current-color"
            title={null}
          >
            <svg
              aria-hidden={true}
              className="slds-icon slds-icon_x-small"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#builder"
              />
            </svg>
          </span>
        </div>
        <div
          className="slds-media__body"
        >
          App Name
        </div>
      </div>
    </div>
    <nav
      className="slds-builder-header__item slds-builder-header__nav"
    >
      <ul
        className="slds-builder-header__nav-list"
      >
        <li
          className="slds-builder-header__nav-item slds-dropdown-trigger slds-dropdown-trigger_click slds-is-open"
        >
          <button
            aria-expanded={false}
            aria-haspopup="true"
            className="slds-button slds-builder-header__item-action slds-media slds-media_center"
            title="Click to open menu"
          >
            <span
              className="slds-media__figure"
            >
              <span
                className="slds-icon_container slds-icon-utility-page slds-current-color"
                title={null}
              >
                <svg
                  aria-hidden={true}
                  className="slds-icon slds-icon_x-small"
                >
                  <use
                    xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#page"
                  />
                </svg>
              </span>
            </span>
            <span
              className="slds-media__body"
            >
              <span
                className="slds-truncate"
                title="Dropdown"
              >
                Dropdown
              </span>
              <span
                className="slds-icon_container slds-icon-utility-chevrondown slds-current-color slds-m-left_small"
                title={null}
              >
                <svg
                  aria-hidden={true}
                  className="slds-icon slds-icon_x-small"
                >
                  <use
                    xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevrondown"
                  />
                </svg>
              </span>
            </span>
          </button>
        </li>
      </ul>
    </nav>
    <div
      className="slds-builder-header__item slds-has-flexi-truncate"
    >
      <h1
        className="slds-builder-header__item-label"
      >
        <span
          className="slds-truncate"
          title="Page Type"
        >
          Page Type
        </span>
      </h1>
    </div>
    <div
      className="slds-builder-header__item slds-builder-header__utilities"
    >
      <div
        className="slds-builder-header__utilities-item"
      >
        <a
          className="slds-builder-header__item-action slds-media slds-media_center"
          href="#"
          onClick={[Function]}
        >
          <span
            className="slds-media__figure"
          >
            <span
              className="slds-icon_container slds-icon-utility-settings slds-current-color"
              title={null}
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon_x-small"
              >
                <use
                  xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
                />
              </svg>
            </span>
          </span>
          <span
            className="slds-media__body"
          >
            <span
              className="slds-truncate"
              title="Link"
            >
              Settings
            </span>
          </span>
        </a>
      </div>
      <div
        className="slds-builder-header__utilities-item"
      >
        <a
          className="slds-builder-header__item-action slds-media slds-media_center"
          href="#"
          onClick={[Function]}
        >
          <div
            className="slds-media__figure"
          >
            <span
              className="slds-icon_container slds-icon-utility-help slds-current-color"
              title={null}
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon_x-small"
              >
                <use
                  xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#help"
                />
              </svg>
            </span>
          </div>
          <div
            className="slds-media__body"
          >
            Help
          </div>
        </a>
      </div>
    </div>
  </header>
  <div
    className="slds-builder-toolbar"
    role="toolbar"
  >
    <div
      aria-label="Canvas Actions"
      className="slds-builder-toolbar__item-group"
    >
      <button
        className="slds-button slds-button_icon slds-button_icon-border"
        onClick={[Function]}
        title="Canvas Settings"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Canvas Settings
        </span>
      </button>
      <div
        className="slds-button-group"
      >
        <button
          className="slds-button slds-button_icon slds-button_icon-border"
          onClick={[Function]}
          tabIndex="0"
          title="Undo"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#undo"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Undo
          </span>
        </button>
        <button
          className="slds-button slds-button_icon slds-button_icon-border"
          onClick={[Function]}
          tabIndex="-1"
          title="Redo"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#redo"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Redo
          </span>
        </button>
      </div>
    </div>
    <div
      aria-label="Edit actions"
      className="slds-builder-toolbar__item-group"
    >
      <div
        className="slds-button-group"
      >
        <button
          className="slds-button slds-button_icon slds-button_icon-border"
          onClick={[Function]}
          tabIndex="-1"
          title="Cut"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#cut"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Cut
          </span>
        </button>
        <button
          className="slds-button slds-button_icon slds-button_icon-border"
          onClick={[Function]}
          tabIndex="-1"
          title="Copy"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#copy"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Copy
          </span>
        </button>
        <button
          className="slds-button slds-button_icon slds-button_icon-border"
          onClick={[Function]}
          tabIndex="-1"
          title="Paste"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#paste"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Paste
          </span>
        </button>
      </div>
      <button
        className="slds-button slds-button_icon slds-button_icon-border"
        onClick={[Function]}
        title="Toggle Panel"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#toggle_panel_left"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Toggle Panel
        </span>
      </button>
    </div>
    <div
      aria-label="Document actions"
      className="slds-builder-toolbar__actions"
    >
      <button
        className="slds-button slds-button_neutral"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_left"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#right"
          />
        </svg>
         
        Run
      </button>
      <button
        className="slds-button slds-button_neutral"
        disabled={false}
        onClick={[Function]}
      >
        Save As
      </button>
      <button
        className="slds-button slds-button_brand"
        disabled={false}
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
</div>
`;

exports[`render builder header render base builder header with toolbar and status with alert 1`] = `
<div
  className="slds-builder-header_container"
>
  <header
    className="slds-builder-header"
  >
    <div
      className="slds-builder-header__item"
    >
      <a
        className="slds-builder-header__item-action"
        href="#"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        title="Back"
      >
        <span
          className="slds-icon_container slds-icon-utility-back slds-current-color"
          title={null}
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon_x-small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#back"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Back
          </span>
        </span>
      </a>
    </div>
    <div
      className="slds-builder-header__item"
    >
      <div
        className="slds-builder-header__item-label slds-media slds-media_center"
      >
        <div
          className="slds-media__figure"
        >
          <span
            className="slds-icon_container slds-icon-utility-builder slds-current-color"
            title={null}
          >
            <svg
              aria-hidden={true}
              className="slds-icon slds-icon_x-small"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#builder"
              />
            </svg>
          </span>
        </div>
        <div
          className="slds-media__body"
        >
          App Name
        </div>
      </div>
    </div>
    <nav
      className="slds-builder-header__item slds-builder-header__nav"
    >
      <ul
        className="slds-builder-header__nav-list"
      >
        <li
          className="slds-builder-header__nav-item slds-dropdown-trigger slds-dropdown-trigger_click slds-is-open"
        >
          <button
            aria-expanded={false}
            aria-haspopup="true"
            className="slds-button slds-builder-header__item-action slds-media slds-media_center"
            title="Click to open menu"
          >
            <span
              className="slds-media__figure"
            >
              <span
                className="slds-icon_container slds-icon-utility-page slds-current-color"
                title={null}
              >
                <svg
                  aria-hidden={true}
                  className="slds-icon slds-icon_x-small"
                >
                  <use
                    xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#page"
                  />
                </svg>
              </span>
            </span>
            <span
              className="slds-media__body"
            >
              <span
                className="slds-truncate"
                title="Dropdown"
              >
                Dropdown
              </span>
              <span
                className="slds-icon_container slds-icon-utility-chevrondown slds-current-color slds-m-left_small"
                title={null}
              >
                <svg
                  aria-hidden={true}
                  className="slds-icon slds-icon_x-small"
                >
                  <use
                    xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevrondown"
                  />
                </svg>
              </span>
            </span>
          </button>
        </li>
      </ul>
    </nav>
    <div
      className="slds-builder-header__item slds-has-flexi-truncate"
    >
      <h1
        className="slds-builder-header__item-label"
      >
        <span
          className="slds-truncate"
          title="Page Type"
        >
          Page Type
        </span>
      </h1>
    </div>
    <div
      className="slds-builder-header__item slds-builder-header__utilities"
    >
      <div
        className="slds-builder-header__utilities-item"
      >
        <a
          className="slds-builder-header__item-action slds-media slds-media_center"
          href="#"
          onClick={[Function]}
        >
          <span
            className="slds-media__figure"
          >
            <span
              className="slds-icon_container slds-icon-utility-settings slds-current-color"
              title={null}
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon_x-small"
              >
                <use
                  xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
                />
              </svg>
            </span>
          </span>
          <span
            className="slds-media__body"
          >
            <span
              className="slds-truncate"
              title="Link"
            >
              Settings
            </span>
          </span>
        </a>
      </div>
      <div
        className="slds-builder-header__utilities-item"
      >
        <a
          className="slds-builder-header__item-action slds-media slds-media_center"
          href="#"
          onClick={[Function]}
        >
          <div
            className="slds-media__figure"
          >
            <span
              className="slds-icon_container slds-icon-utility-help slds-current-color"
              title={null}
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon_x-small"
              >
                <use
                  xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#help"
                />
              </svg>
            </span>
          </div>
          <div
            className="slds-media__body"
          >
            Help
          </div>
        </a>
      </div>
    </div>
  </header>
  <div
    className="slds-builder-toolbar"
    role="toolbar"
  >
    <div
      aria-label="Canvas Actions"
      className="slds-builder-toolbar__item-group"
    >
      <button
        className="slds-button slds-button_icon slds-button_icon-border"
        onClick={[Function]}
        title="Canvas Settings"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Canvas Settings
        </span>
      </button>
      <div
        className="slds-button-group"
      >
        <button
          className="slds-button slds-button_icon slds-button_icon-border"
          onClick={[Function]}
          tabIndex="0"
          title="Undo"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#undo"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Undo
          </span>
        </button>
        <button
          className="slds-button slds-button_icon slds-button_icon-border"
          onClick={[Function]}
          tabIndex="-1"
          title="Redo"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#redo"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Redo
          </span>
        </button>
      </div>
    </div>
    <div
      aria-label="Edit actions"
      className="slds-builder-toolbar__item-group"
    >
      <div
        className="slds-button-group"
      >
        <button
          className="slds-button slds-button_icon slds-button_icon-border"
          onClick={[Function]}
          tabIndex="-1"
          title="Cut"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#cut"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Cut
          </span>
        </button>
        <button
          className="slds-button slds-button_icon slds-button_icon-border"
          onClick={[Function]}
          tabIndex="-1"
          title="Copy"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#copy"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Copy
          </span>
        </button>
        <button
          className="slds-button slds-button_icon slds-button_icon-border"
          onClick={[Function]}
          tabIndex="-1"
          title="Paste"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#paste"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Paste
          </span>
        </button>
      </div>
      <button
        className="slds-button slds-button_icon slds-button_icon-border"
        onClick={[Function]}
        title="Toggle Panel"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#toggle_panel_left"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Toggle Panel
        </span>
      </button>
    </div>
    <div
      aria-label="Document actions"
      className="slds-builder-toolbar__actions"
    >
      <button
        aria-live="polite"
        className="slds-button slds-p-horizontal_small"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        Inactive - Modified 1 day ago
      </button>
      <button
        className="slds-button slds-button_icon slds-button_icon-container slds-button_icon-warning"
        onBlur={[Function]}
        onClick={[Function]}
        title="warning"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#warning"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          warning
        </span>
      </button>
      <button
        className="slds-button slds-button_icon slds-button_icon-container slds-button_icon-error"
        onBlur={[Function]}
        onClick={[Function]}
        title="error"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#error"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          error
        </span>
      </button>
      <button
        className="slds-button slds-button_neutral"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-button__icon_left"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#right"
          />
        </svg>
         
        Run
      </button>
      <button
        className="slds-button slds-button_neutral"
        disabled={false}
        onClick={[Function]}
      >
        Save As
      </button>
      <button
        className="slds-button slds-button_brand"
        disabled={false}
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
</div>
`;

exports[`render builder header render base builder header with truncated name 1`] = `
<div
  className="slds-builder-header_container"
>
  <header
    className="slds-builder-header"
  >
    <div
      className="slds-builder-header__item"
    >
      <a
        className="slds-builder-header__item-action"
        href="#"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        title="Back"
      >
        <span
          className="slds-icon_container slds-icon-utility-back slds-current-color"
          title={null}
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon_x-small"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#back"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Back
          </span>
        </span>
      </a>
    </div>
    <div
      className="slds-builder-header__item"
    >
      <div
        className="slds-builder-header__item-label slds-media slds-media_center"
      >
        <div
          className="slds-media__figure"
        >
          <span
            className="slds-icon_container slds-icon-utility-builder slds-current-color"
            title={null}
          >
            <svg
              aria-hidden={true}
              className="slds-icon slds-icon_x-small"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#builder"
              />
            </svg>
          </span>
        </div>
        <div
          className="slds-media__body"
        >
          App Name
        </div>
      </div>
    </div>
    <nav
      className="slds-builder-header__item slds-builder-header__nav"
    >
      <ul
        className="slds-builder-header__nav-list"
      >
        <li
          className="slds-builder-header__nav-item slds-dropdown-trigger slds-dropdown-trigger_click slds-is-open"
        >
          <button
            aria-expanded={false}
            aria-haspopup="true"
            className="slds-button slds-builder-header__item-action slds-media slds-media_center"
            title="Click to open menu"
          >
            <span
              className="slds-media__figure"
            >
              <span
                className="slds-icon_container slds-icon-utility-page slds-current-color"
                title={null}
              >
                <svg
                  aria-hidden={true}
                  className="slds-icon slds-icon_x-small"
                >
                  <use
                    xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#page"
                  />
                </svg>
              </span>
            </span>
            <span
              className="slds-media__body"
            >
              <span
                className="slds-truncate"
                title="Dropdown"
              >
                Dropdown
              </span>
              <span
                className="slds-icon_container slds-icon-utility-chevrondown slds-current-color slds-m-left_small"
                title={null}
              >
                <svg
                  aria-hidden={true}
                  className="slds-icon slds-icon_x-small"
                >
                  <use
                    xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevrondown"
                  />
                </svg>
              </span>
            </span>
          </button>
        </li>
      </ul>
    </nav>
    <div
      className="slds-builder-header__item slds-has-flexi-truncate"
    >
      <h1
        className="slds-builder-header__item-label"
      >
        <span
          className="slds-truncate"
          title="Page Type with a very very long name that will truncate when the container is not wide enough to contain this content completely"
        >
          Page Type with a very very long name that will truncate when the container is not wide enough to contain this content completely
        </span>
      </h1>
    </div>
    <div
      className="slds-builder-header__item slds-builder-header__utilities"
    >
      <div
        className="slds-builder-header__utilities-item"
      >
        <a
          className="slds-builder-header__item-action slds-media slds-media_center"
          href="#"
          onClick={[Function]}
        >
          <span
            className="slds-media__figure"
          >
            <span
              className="slds-icon_container slds-icon-utility-settings slds-current-color"
              title={null}
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon_x-small"
              >
                <use
                  xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
                />
              </svg>
            </span>
          </span>
          <span
            className="slds-media__body"
          >
            <span
              className="slds-truncate"
              title="Link"
            >
              Settings
            </span>
          </span>
        </a>
      </div>
      <div
        className="slds-builder-header__utilities-item"
      >
        <a
          className="slds-builder-header__item-action slds-media slds-media_center"
          href="#"
          onClick={[Function]}
        >
          <div
            className="slds-media__figure"
          >
            <span
              className="slds-icon_container slds-icon-utility-help slds-current-color"
              title={null}
            >
              <svg
                aria-hidden={true}
                className="slds-icon slds-icon_x-small"
              >
                <use
                  xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#help"
                />
              </svg>
            </span>
          </div>
          <div
            className="slds-media__body"
          >
            Help
          </div>
        </a>
      </div>
    </div>
  </header>
</div>
`;
